test/pickfirst_test.go

```diff
@@ -346,40 +351,33 @@ func (s) TestPickFirst_StickyTransientFailure(t *testing.T) {
 	}
 	t.Cleanup(func() { cc.Close() })
 
-	var wg sync.WaitGroup
-	wg.Add(2)
-	// Spin up a goroutine that waits for the channel to move to
-	// TransientFailure. After that it checks that the channel stays in
-	// TransientFailure, until Shutdown.
-	go func() {
-		defer wg.Done()
-		for state := cc.GetState(); state != connectivity.TransientFailure; state = cc.GetState() {
-			if !cc.WaitForStateChange(ctx, state) {
-				t.Errorf("Timeout when waiting for state to change to TransientFailure. Current state is %s", state)
-				return
-			}
+	// Wait for the channel to move to TransientFailure.
+	for state := cc.GetState(); state != connectivity.TransientFailure; state = cc.GetState() {
```

Comment:
Please use [awaitState](https://github.com/grpc/grpc-go/blob/f0280f9d3d2f3fbe45de1ab5873a19ec678c74c8/test/clientconn_state_transition_test.go#L523) instead.  Then you don't really need the comment.

Reply:
Done.